### PR TITLE
fix(rest): Added component type tag in release api

### PR DESCRIPTION
**Pull request by nikkuma7**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? #1607 
> * Did you add or update any new dependencies that are required for your change? yes, I added component type tag under release API.

Issue: 

1) Go to component tab and click on any component name.
2) If you need to change in the exiting component or you can select Release Overview tab.
3)click on version and copy release Id and test it in postman.
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Testing was successful but only for Getting a single release by REST.
Component type field not showing for other rest call related to releases.

**Comment by rudra-superrr** [comment]
Testing was successful.

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java [hunk 1]
```diff
@@ -454,6 +455,7 @@ public void should_document_get_release() throws Exception {
                                 fieldWithPath("cpeId").description("The CPE id"),
                                 fieldWithPath("releaseDate").description("The date of this release"),
                                 fieldWithPath("createdOn").description("The creation date of the internal sw360 release"),
+                                fieldWithPath("componentType").description("The componentType of the release"),
```
Comment (smrutis1) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java:458]: Add all possible values

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java [hunk 2]
```diff
@@ -732,6 +733,7 @@ private RestDocumentationResultHandler documentReleaseProperties() {
                         fieldWithPath("clearingState").description("The clearing of the release, possible values are " + Arrays.asList(ClearingState.values())),
                         fieldWithPath("cpeId").description("The CPE id"),
                         fieldWithPath("releaseDate").description("The date of this release"),
+                        fieldWithPath("componentType").description("The componentType of the release"),
```
Comment (smrutis1) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java:736]: Add all possible values

**Review by smrutis1 — CHANGES_REQUESTED** [review]

**Review by smrutis1 — APPROVED** [review]
Code looks good.

**Pull request merged**